src/ray/raylet/node_manager.cc

```diff
@@ -1518,11 +1529,16 @@ void NodeManager::NewSchedulerSchedulePendingTasks() {
       /// There is no node that has available resources to run the request.
       break;
     } else {
-      new_resource_scheduler_->SubtractNodeAvailableResources(node_id_string,
-                                                              request_resources);
       if (node_id_string == client_id_.Binary()) {
-        tasks_to_dispatch_.push_back(work);
+        if (work.second.GetTaskSpecification().GetDependencies().size() == 0) {
+          tasks_to_dispatch_.push_back(work);
+        } else {
+          WaitForTaskArgsRequests(work);
```

Comment:
Consider always calling WaitForTaskArgsRequests even if deps size is zero.

Reply:
ok